packages/caravan-fees/src/btcTransactionTemplate.ts

```diff
@@ -30,12 +41,90 @@ export class BtcTransactionTemplate {
     this._inputs = options.inputs || [];
     this._outputs = options.outputs || [];
     this._targetFeeRate = options.targetFeeRate;
-    this._dustThreshold = options.dustThreshold;
+    this._dustThreshold = new BigNumber(
+      options.dustThreshold || DEFAULT_DUST_THRESHOLD_IN_SATS,
+    );
     this._network = options.network;
     this._scriptType = options.scriptType;
     this._requiredSigners = options.requiredSigners || 1;
     this._totalSigners = options.totalSigners || 1;
   }
+
+  /**
+   * Creates a BtcTransactionTemplate from a raw PSBT hex string.
+   * This method parses the PSBT, extracts input and output information,
+   * and creates a new BtcTransactionTemplate instance.
+   *
+   * @param psbtHex - The raw PSBT hex string
+   * @param options - Additional options for creating the template
+   * @returns A new BtcTransactionTemplate instance
+   * @throws Error if PSBT parsing fails or required information is missing
+   */
+  static fromRawPsbt(
+    psbtHex: string,
```

Comment:
Maybe just “rawPsbt” because I think the initialize function supports, hex, baee64, and buffer (but please confirm). Either way that’s preferable for the api imo

Reply:
Yes `fromRawPsbt` does suppoort initialisation from any of the above said types

changes [done](https://github.com/caravan-bitcoin/caravan/pull/114/commits/6662b2d56bc5f1cef57c403cdc13c2a0ccaac980#diff-c113a33821d4e86e4ad4f20743227a51337831792cc5c5926aa2a5971c11d7b1R65)

Reply:
I think this variable name still needs to be changed to `rawPsbt` since this supports initializing from not just hex

Reply:
it's already named [`fromPsbt`](https://github.com/caravan-bitcoin/caravan/pull/114/commits/945d644baf5fcfbfac83271deebe440262046414#diff-c113a33821d4e86e4ad4f20743227a51337831792cc5c5926aa2a5971c11d7b1R64) .... I guess you are looking at a older commit

Reply:
Ah ok. Definitely likely! 😅